### DISCUSSION TITLE
Sudden change in total stars earned

### DISCUSSION THREAD

**Discussion by andreped** [anuraghazra/github-readme-stats#2186] [Q&A] [answered]
Just observed that the total number of stars just changed.

As this is not really an issue, as this change was _probably_ done intentional in a recent commit, for whatever reason, I would like to know if there is a way to get the old behaviour. Just went from above 300 stars to around 136 (see https://github.com/andreped).

Any suggestions?

**Comment by rickstaa** [comment, marked as answer]
@andreped to my knowledge, we did not make breaking changes to the star fetching behaviour. 🤔 The only commit that could have influenced this would be af97e5765bea4945e3208e63d272248b91da2b3c. I checked your username in the [GraphQL API](https://docs.github.com/en/graphql/overview/explorer), and the correct number of stars should be 136. You can check this out using the following syntax:

```graphql
query userInfo($login: String!) {
  user(login: $login) {
    repositories(
      first: 100
      ownerAffiliations: OWNER
      orderBy: {direction: DESC, field: STARGAZERS}
    ) {
      nodes {
        name
        stargazers {
          totalCount
        }
      }
    }
  }
}
```

```graphql
{
  "login": "andreped"
}
```

I am, therefore, surprised that you used to have 300 stars. Did you maybe transfer a repository with stars to an organization? We are currently not counting stars from repositories that are in organizations (see #1808).

I also checked the commit before this change, resulting in 136 stars. You can check this out yourself by:
1. cloning the repository `git clone git@github.com:anuraghazra/github-readme-stats.git`.
2. Installing the npm packages: `npm i`.
3. Checking out older commits `git reset --hard acbc03dc0f2e70da946fbb75822bbe9d041fb729`.
4. Starting the development server `vercel dev`.
5. Check out your username in the browser `http://localhost:3000/api?username=andreped`.

Please let me know if you find a commit where the stars are returned as 300.

**Reply by andreped** [reply]
I believe what has changed is that repos where I am collaborator have not been taken into account anymore, which they used to be.

**Reply by rickstaa** [reply]
This was changed more than 2 years ago in https://github.com/anuraghazra/github-readme-stats/commit/96d37c706fd0d1df0845ac09763cde026e515c1b (see #39 for why this was done). Before that commit, the `ownerAffiliations: OWNER` parameter was not present. You can see the results, which is 305 stars, by using:

```graphql
query userInfo($login: String!) {
  user(login: $login) {
    repositories(
      first: 100
      orderBy: {direction: DESC, field: STARGAZERS}
    ) {
      nodes {
        name
        stargazers {
          totalCount
        }
      }
    }
  }
}
```

This change was made since, due to GraphQL and Vercel timeout limitations, we can not support fetching repositories of which the user is not the owner (See #1471). We are planning to bring this back for self-deployed Vercel instances. For please check out https://github.com/anuraghazra/github-readme-stats/discussions/1808 on how you can already use this feature.

**Reply by rickstaa** [reply]
@andreped On second thought, the current behaviour makes sense. Users should not, by default, receive stars of repositories of which they simply are an `ORGANIZATION_MEMBER`. For the `COLLABORATOR`, it is a different story adding this feature is discussed in #1.

**Reply by andreped** [reply]
This changed just a few days ago. It was working before. Not sure why it stopped working as intended.

**Reply by rickstaa** [reply]
@andreped I am not able to reproduce your issue. So, unfortunately, I can not help you any further. If you find the commit that you think caused this behaviour, feel free to tag me. For now, if you want the 300 stars to show up, please follow my steps at https://github.com/anuraghazra/github-readme-stats/issues/1#issuecomment-981419174.

**Reply by andreped** [reply]
Apparently, adding this `&role=OWNER,ORGANIZATION_MEMBER,COLLABORATOR` does nothing anymore. I believe this had an effect before. That's probably why the stars are gone.

**Reply by rickstaa** [reply]
@andreped Ah, I now see what happened. You are using my Vercel instance instead of the public one. 😅 I made some breaking changes some days ago but reverted them somewhere yesterday. That is why the `&role` parameter stopped working for you. Can you please accept my answer above so we can mark this issue as resolved 🤓?

**Reply by andreped** [reply]
> @andreped Ah, I now see what happened. You are using my Vercel instance instead of the public one. 😅 I made some breaking changes some days ago but reverted them somewhere yesterday. That is why the `&role` parameter stopped working for you. Can you please accept my answer above so we can mark this issue as resolved 🤓?

I just changed to use your vercel instance yesterday. Previously I used the correct one, which was when I got the wrong number of stars.

No idea what the problem is. But you can solve it now as I have a temporary fix at least. If others observe the same, we can reopen this Discussion.

But thanks for trying :]
